### ISSUE TITLE
Sidecar build with image URL broken after update

### ISSUE THREAD

**Issue by araujogo** [aws/copilot-cli#4708]
I'm not able to deploy the service after the 1.27 upgrade. Changelog did't specify any breaking changes and Documentation of the original sidecar patterns still alive. So this configurations:
```
sidecars:
  fluent-bit:
    port: 5170
    image: public.ecr.aws/calculato/custom-fluent-bit:v1.2
```
Shouldn't be failing with
`
 deploy service conta to environment test: deploy service: check if changeset is empty: create change set copilot-3161a968-20d4-42ef-a0b4-5c6719ea0372 for stack ommited: ValidationError: Template format error: Parameter name fluent-bitEnvFileARN is non alphanumeric.
`

**Comment by paragbhingre** [comment]
Hi @araujogo I am so sorry to hear that you ran into this problem. We recently added a new feature to add env files support for all sidecar containers. As a part of that addition, we potentially introduced a new bug here, as CFN doesn't allow using special characters in the resource names, and as you can see, your sidecar fluent-bit has a `-` in it. I am fixing this one, but in the meantime, if you could just remove `-` from your sidecar name, then everything will work fine. 

Keep us posted if workaround works fine for you.

**Comment by araujogo** [comment]
That worked, thank you!

**Comment by KollaAdithya** [comment]
Fix is now released in v1.28 https://github.com/aws/copilot-cli/releases/tag/v1.28.0. Thanks @araujogo for reporting this!
